src/main/java/at/ac/fhcampuswien/fhmdb/HomeController.java

```diff
@@ -13,104 +13,160 @@
+        releaseYearPicker.getItems().addAll(Year.values());
+        releaseYearPicker.setValue(Year.NO_FILTER);
 
-        sortBtn.setOnAction(actionEvent -> {
-            if(sortBtn.getText().equals("Sort (asc)")) {
-                observableMovies.sort(Comparator.naturalOrder());
-                sortBtn.setText("Sort (desc)");
-            } else {
-                observableMovies.sort(Collections.reverseOrder());
-                sortBtn.setText("Sort (asc)");
-            }
-        });
+        ratingComboBox.getItems().addAll(Rating.values());
+        ratingComboBox.setValue(Rating.NO_FILTER);
 
-        resetFilterBtn.setOnAction(actionEvent -> resetFilterCriteria(genreComboBox, filteredList, searchField));
+        searchBtn.setOnAction(actionEvent -> setFilter());
+        sortBtn.setOnAction(actionEvent -> sortMoviesAndSetButtonText());
+        resetFilterBtn.setOnAction(actionEvent -> resetFilter());
     }
 
-    public static void resetFilterCriteria(JFXComboBox<Genre> genreComboBox,
-                                           FilteredList<Movie> filteredList,
-                                           TextField searchField) {
+    private void resetFilter() {
+        observableMovies.clear();
         searchField.clear();
-        genreComboBox.setValue(Genre.ALL);
-        if (genreComboBox.getValue() != null) {
-            genreComboBox.getSelectionModel().clearSelection();
+
+        sortBtn.setText(SORT_DEFAULT_TEXT_DESC);
+
+        genreComboBox.setValue(Genre.NO_FILTER);
+        releaseYearPicker.setValue(Year.NO_FILTER);
+        ratingComboBox.setValue(Rating.NO_FILTER);
+
+        List<Movie> allMovies;
+
+        try {
+            allMovies = MovieAPIService.getMovies();
+        } catch (IOException e){
+            allMovies = new ArrayList<>();
         }
-        filteredList.setPredicate(movie -> true);
+
+        observableMovies.addAll(allMovies);
+        sortMovies();
     }
 
-    public static void searchForMovie(String searchTerm, Genre genre, FilteredList<Movie> filteredList, List<Movie> movies) {
-        Set<Movie> searchResults = new HashSet<>();
-        Set<Movie> keywordSearchResults = MovieSearchService.searchKeyword(searchTerm, movies);
-        List<Movie> genreSearchResults = MovieFilterService.filterMoviesByGenre(genre, movies);
+    private void setFilter() {
+        observableMovies.clear();
+
+        List<Movie> moviesWithFilter;
 
-        for (Movie movie : keywordSearchResults) {
-            if (genreSearchResults.contains(movie)) {
-                searchResults.add(movie);
-            }
+        String genre = genreComboBox.getValue() == Genre.NO_FILTER ? "" : genreComboBox.getValue().name();
+        String releaseYear = releaseYearPicker.getValue() == Year.NO_FILTER ? "" : releaseYearPicker.getValue().toString();
+        String ratingFrom = ratingComboBox.getValue() == Rating.NO_FILTER ? "" : String.valueOf(ratingComboBox.getValue().getRatingFrom());
```

Comment:
You are using `""` several times, it could be saved in a variable, say `final String NO_FILTER = ""`

Reply:
Ah no, it is like this.

String releaseYear;
if(releaseYearPicker.getValue() == Year.NO_FILTER){
   releaseYear = "";
}else{
   releaseYear = releaseYearPicker.getValue().toString();
}

But I could rewrite, so that it is easier to read.

Reply:
Yes, I understood that it's the short version of the if-else statement, but I was saying that you use a string several times, it would be good to extract it as a constant

Reply:
There is a NO_FILTER constant in HomeController now.